File: src/AJKIOT.Shared/Models/DeviceFeatures/DeviceFeature.cs
using AJKIOT.Shared.Enums;

namespace AJKIOT.Shared.Models.DeviceFeatures
{
    public class DeviceFeature
    {
        public int Id { get; set; }
        public string Type { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public int Value { get; set; }
        public int MinValue { get; set; }
        public int MaxValue { get; set; }
        public int NumberOfSteps { get; set; }

        public bool IsScheduleAvailable()
        {
            var scheduledTypes = Enum.GetNames(typeof(ScheduledFeature)).ToList();
            if (scheduledTypes is null)
                return false;
            return scheduledTypes.Contains(Type);
        }
        public void Update(DeviceFeature deviceFeature)
        {
            Type = deviceFeature.Type;
            Name = deviceFeature.Name;
            Value = deviceFeature.Value;
            MinValue = deviceFeature.MinValue;
            MaxValue = deviceFeature.MaxValue;
            NumberOfSteps = deviceFeature.NumberOfSteps;
        }
    }
}

File: src/ajk_iot_mobile/lib/models/device_feature.dart
class DeviceFeature {
  int id;
  String type;
  String name;
  int value;
  int minValue;
  int maxValue;
  int numberOfSteps;

  DeviceFeature({
    required this.id,
    required this.type,
    required this.name,
    required this.value,
    required this.minValue,
    required this.maxValue,
    required this.numberOfSteps,
  });

  Map<String, dynamic> toJson() => {
    'Id': id,
    'Type': type,
    'Name': name,
    'Value': value,
    'MinValue': minValue,
    'MaxValue': maxValue,
    'NumberOfSteps': numberOfSteps,
  };

  static DeviceFeature fromJson(Map<String, dynamic> json) => DeviceFeature(
    id: json['Id'],
    type: json['Type'],
    name: json['Name'],
    value: json['Value'],
    minValue: json['MinValue'],
    maxValue: json['MaxValue'],
    numberOfSteps: json['NumberOfSteps'],
  );
}